universal-login-react/src/ui/TopUp/Fiat/TopUpWithFiat.tsx

```diff
@@ -93,7 +93,10 @@ export const TopUpWithFiat = ({sdk, onPayClick, logoColor}: TopUpWithFiatProps)
         <button
           onClick={() => onPayClick(paymentMethod!, amount)}
           className="pay-btn"
-          disabled={!paymentMethod || !amount}
+          disabled={
+            !paymentMethod ||
+            (topUpProviderSupportService.isInputAmountUsed(paymentMethod) && !amount)
```

Comment:
Maybe extract this condition to separate function

Reply:
Done.